tasks/main.yml

```diff
@@ -17,26 +12,328 @@
+      service:
+        name: mssql-server
+        state: started
+
+    - name: Check if a custom tcpport setting exist
+      command: /opt/mssql/bin/mssql-conf get network tcpport
+      ignore_errors: true
+      changed_when: false
+      register: __mssql_custom_tcp_port
+
+    - name: Check if a custom ipaddress setting exist
+      command: /opt/mssql/bin/mssql-conf get network ipaddress
+      ignore_errors: true
+      changed_when: false
+      register: __mssql_custom_ip_address
+
+    - name: >-
+        Check if the password matches the existing password.
+        When IP address and TCP port are default.
+      command: >-
+        /opt/mssql-tools/bin/sqlcmd -U sa
+        -P '{{ mssql_password }}' -Q 'SELECT @@VERSION'
+      register: __mssql_sqlcmd_query_default_settings
+      ignore_errors: true
+      changed_when: false
+      when:
+        - '"ipaddress" not in __mssql_custom_ip_address.stdout'
+        - '"tcpport" not in __mssql_custom_tcp_port.stdout'
+
+    - name: >-
+        Check the password matches the existing password.
+        When IP address is custom and TCP port is default.
+      command: >-
+        /opt/mssql-tools/bin/sqlcmd
+        -S {{ __mssql_custom_ip_address.stdout | regex_search('[^: ]*$') }}
+        -U sa -P '{{ mssql_password }}' -Q 'SELECT @@VERSION'
+      register: __mssql_sqlcmd_query_custom_ip
+      ignore_errors: true
+      changed_when: false
+      when:
+        - '"ipaddress" in __mssql_custom_ip_address.stdout'
+        - '"tcpport" not in __mssql_custom_tcp_port.stdout'
+
+    - name: >-
+        Check the password matches the existing password.
+        When TCP port is custom and IP address is default.
+      command: >-
+        /opt/mssql-tools/bin/sqlcmd -S
+        localhost,{{ __mssql_custom_tcp_port.stdout | regex_search('[^: ]*$') }}
+        -U sa -P '{{ mssql_password }}' -Q 'SELECT @@VERSION'
+      register: __mssql_sqlcmd_query_custom_tcp_port
+      ignore_errors: true
+      changed_when: false
+      when:
+        - '"ipaddress" not in __mssql_custom_ip_address.stdout'
+        - '"tcpport" in __mssql_custom_tcp_port.stdout'
+
+    - name: >-
+        Check the password matches the existing password.
+        When the IP address and TCP port settings are custom.
+      command: >-
+        /opt/mssql-tools/bin/sqlcmd
+        -S {{ __mssql_custom_ip_address.stdout |
+        regex_search('[^: ]*$') }},{{ __mssql_custom_tcp_port.stdout |
+        regex_search('[^: ]*$') }}
+        -U sa -P '{{ mssql_password }}' -Q 'SELECT @@VERSION'
+      register: __mssql_sqlcmd_query_custom_settings
+      ignore_errors: true
+      changed_when: false
+      when:
+        - '"ipaddress" in __mssql_custom_ip_address.stdout'
+        - '"tcpport" in __mssql_custom_tcp_port.stdout'
+
+    - name: Ensure that the mssql-server service is stopped
+      service:
+        name: mssql-server
+        state: stopped
+      when: >-
```

Comment:
I recommend using another `set_fact` for this so you can reuse it below

Reply:
I used your solution with `ternary` so this when is not needed